src/getfields.jl

```diff
@@ -214,7 +214,7 @@ function parsefield(io::Union{IOBuffer,UnsafeBuffer}, ::Type{Date}, opt::CSV.Opt
                 break
             end
         end
-        val = String(pointer(io.data)+pos, position(io)-pos-quoted-1+end_of_file)
+        val = Compat.bytestring(pointer(io.data)+pos, position(io)-pos-quoted-1+end_of_file)
```

Comment:
Sorry, I think I was mistaken. According to Compat's README, this should be `@compat String(...)`, and same below.